rio_tiler/io/base.py

```diff
@@ -527,6 +600,7 @@ def _reader(asset: str, *args: Any, **kwargs: Any) -> ImageData:
         if expression:
             blocks = expression.split(",")
             output.data = apply_expression(blocks, assets, output.data)
+            output.band_names = blocks
```

Comment:
here we set the `band_names` to be the name of the expression blocks. This will lose any information about indexes or expression within assets. e.g if you pass `expression="red/blue", indexes=2` or `expression="red/blue", asset_expression=b1*5`, `band_names` will be set to `red/blue`

Reply:
`expression` will be mostly used for STAC with `one band` so I think for now it's ok